packages/flutter/lib/src/widgets/linked_text.dart

```diff
@@ -0,0 +1,776 @@
+    super.key,
+    this.builder = defaultBuilder,
+    required LinkTapCallback onTap,
+    required RegExp regExp,
+    List<InlineSpan>? spans,
+    String? text,
+  }) : assert((text == null) != (spans == null), 'Must specify exactly one to link: either text or spans.'),
+       textLinkers = <TextLinker>[
+         TextLinker(
+           textRangesFinder: TextLinker.textRangesFinderFromRegExp(regExp),
+           linkBuilder: getDefaultLinkBuilder(onTap),
+         ),
+       ],
+       spans = spans ?? <InlineSpan>[
+         TextSpan(
+           text: text,
+         ),
+       ];
+
+  /// Creates an instance of [LinkedText] where the given [textLinkers] are
+  /// applied.
+  ///
+  /// {@template flutter.widgets.LinkedText.textLinkers}
+  /// Useful for independently matching different types of strings with
+  /// different behaviors. For example, highlighting both URLs and Twitter
+  /// handles with different style and/or behavior.
+  /// {@endtemplate}
+  ///
+  /// The [TextLinker.textRangesFinder] must not produce overlapping
+  /// [TextRange]s.
+  ///
+  /// {@tool dartpad}
+  /// This example shows how to use [LinkedText.textLinkers] to link both URLs
+  /// and Twitter handles independently.
+  ///
+  /// ** See code in examples/api/lib/widgets/linked_text/linked_text.2.dart **
+  /// {@end-tool}
+  ///
+  /// See also:
+  ///
+  ///  * [LinkedText.new], which can be passed [TextRange]s directly or
+  ///    otherwise matches URLs by default.
+  ///  * [LinkedText.regExp], which automatically finds ranges that match
+  ///    the given [RegExp].
+  LinkedText.textLinkers({
+    super.key,
+    this.builder = defaultBuilder,
+    String? text,
+    List<InlineSpan>? spans,
+    required this.textLinkers,
+  }) : assert((text == null) != (spans == null), 'Must specify exactly one to link: either text or spans.'),
+       assert(textLinkers.isNotEmpty),
+       spans = spans ?? <InlineSpan>[
+         TextSpan(
+           text: text,
+         ),
+       ];
+
+  static final RegExp _urlRegExp = RegExp(r'(?<!@[a-zA-Z0-9-]*)(?<![\/\.a-zA-Z0-9-])((https?:\/\/)?(([a-zA-Z0-9-]*\.)*[a-zA-Z0-9-]+(\.[a-zA-Z]+)+))(?::\d{1,5})?(?:\/[^\s]*)?(?:\?[^\s#]*)?(?:#[^\s]*)?(?![a-zA-Z0-9-]*@)');
+
+  /// A [TextRangesFinder] that returns [TextRange]s for URLs.
+  ///
+  /// Matches full (https://www.example.com/?q=1) and shortened (example.com)
+  /// URLs.
+  ///
+  /// Excludes:
+  ///
+  ///   * URLs with any protocol other than http or https.
+  ///   * Email addresses.
+  static final TextRangesFinder defaultTextRangesFinder = TextLinker.textRangesFinderFromRegExp(_urlRegExp);
+
+  /// The spans on which to create links by applying [textLinkers].
+  final List<InlineSpan> spans;
+
+  /// Defines what parts of the text to match and how to link them.
+  ///
+  /// [TextLinker]s are applied in the order given. Overlapping matches are not
+  /// supported and will produce an error.
+  late final List<TextLinker> textLinkers;
```

Comment:
why late?

Reply:
I must have forgotten this from an older refactor, good catch.